wordcab_transcribe/main.py

```diff
@@ -131,18 +133,20 @@ async def inference_with_audio(
     else:
         filepath = filename
 
-    utterances = await asr.process_input(
+    raw_utterances = await asr.process_input(
         filepath, num_speakers, source_lang, timestamps
     )
-    utterances = [
-        {
-            "text": str(utterance["text"]),
-            "start": utterance["start"],
-            "end": utterance["end"],
-            "speaker": int(utterance["speaker"]),
-        }
-        for utterance in utterances
-    ]
+    utterances = []
+    for utterance in raw_utterances:
+        if not is_empty_string(utterance["text"]):
+            utterances.append(
+                {
+                    "text": format_punct(utterance["text"]),
+                    "start": utterance["start"],
+                    "end": utterance["end"],
+                    "speaker": int(utterance["speaker"]),
+                }
+            )
```

Comment:
I would keep the list comprehension style for readability.

```suggestion
    utterances = [
        {
            "text": format_punct(utterance["text"]),
            "start": utterance["start"],
            "end": utterance["end"],
            "speaker": int(utterance["speaker"]),
        }
        for utterance in raw_utterances
        if not is_empty_string(utterance["text"])
    ]
```